sklego/meta/estimator_transformer.py

```diff
@@ -38,6 +39,36 @@ def transform(self, X):
         Returns array of shape `(X.shape[0], )` if estimator is not multi output.
         For multi output estimators an array of shape `(X.shape[0], y.shape[1])` is returned.
         """
-        check_is_fitted(self, "estimator_")
```

Comment:
Why are we no longer checking `estimator_`?

Reply:
I moved this call to `__sklearn_is_fitted`, so the behavior hasn't changed.

`check_is_fitted` will call `self.__sklearn_is_fitted` ([sklearn source](https://github.com/scikit-learn/scikit-learn/blob/626b4608d4f840af7c37bff2ccb38fcfd2ef594f/sklearn/utils/validation.py#L1338), which contains `check_is_fitted(self.estimator_)`. 

Not sure if this obscures the behavior too much. Implementing `__sklearn_is_fitted` works well for sklearn transformers, but I can imagine not that many people are familiar with it.

Reply:
I'm cool with the implementation, though I'd prefer to have a comment around explaining it. Should their internals change, we'll have a link.

Reply:
Sounds good! Added a comment for clarification.

Reply:
Could you also add a link to the sklearn source? Also; since this is a hidden method; comments may be better, otherwise tools like sphinx might accidentally document it.

Reply:
Sure! Turned some lines into comments and added link to exact line where `__sklearn_is_fitted` is called.